### DISCUSSION TITLE
<script setup> interfering with v-model in production

### DISCUSSION THREAD

**Discussion by pschnyder** [nuxt/nuxt#17160] [Questions]
Hello, I have the following issue on a Nuxt 3 project I'm trying to deploy to Vercel (also tried Netflify and got the same error):

In my Vue components I am using the Options API but sometimes also `<script setup>` to define route middleware. This works very well in dev mode, however, on production v-model doesn't work anymore and as soon as I delete the  `<script setup>` it works again.

I get the following error on any change to the input that uses v-model ("newStartingMonth" is the variable in v-model):

<img width="480" alt="Bildschirmfoto 2022-07-30 um 12 03 37" src="https://user-images.githubusercontent.com/69150941/181908038-42f5f8ec-22d3-443e-adaf-b5123a4677aa.png">

As the compositions API is pretty new to me, I assume that I'm not allowed to use the `<script setup>` tag alongside a "normal" component using the options API? However, if that's the case shouldn't it also not work on dev?

Thank you!

**Comment by pschnyder** [comment]
UPDATE: I changed the following and now it works!

From:
```
[...]
v-model="startingBalance"
[...]
data() {
    return {
      startingBalance: null
   }
}
```
To:
```
[...]
v-model="startingBalance.value"
[...]
data() {
    return {
      startingBalance: {
        value: null
      },
   }
}
```
Is that expected behaviour? If yes, why would it work on dev but not in production?

**Comment by danielroe** [comment]
This is Vue behaviour. See https://github.com/nuxt/framework/issues/6065#issuecomment-1192530314.
